x/reward/abci.go

```diff
@@ -77,13 +87,11 @@ func handleExternalChainVotingInflation(ctx sdk.Context, k types.Rewarder, n typ
 			continue
 		}
 
-		reward := sdk.NewCoin(
-			denom,
-			amountPerChain.QuoInt64(int64(len(maintainers))).RoundInt(),
-		)
-		for _, maintainer := range maintainers {
-			// Each maintainer receives equal amount of reward
-			rewardPool.AddReward(maintainer, reward)
+		validators := make([]stakingtypes.Validator, len(maintainers))
+		for i, maintainer := range maintainers {
+			validators[i] = s.Validator(ctx, maintainer).(stakingtypes.Validator)
```

Comment:
`s.Validator` could return `nil` if the maintainer isn't a validator anymore?
Also, I'm looking at the ante handle for `MsgUndelegate` and while we don't allow a validator to unbond while holding TSS key shares, we don't seem to check if the validator proxy's was deactivated first (so it seems a validator could unbond without deactivating their proxy). Is this intentional?

Reply:
@milapsheth 

"s.Validator could return nil if the maintainer isn't a validator anymore?" Will check.
"we don't seem to check if the validator proxy's was deactivated first" Why should we?

Reply:
Wouldn't we like to auto-deactivate the proxy address if the operator isn't a validator anymore? We would rely on inner handler checking to make sure they're still a validator. Or do we want the proxy account concept to be separate from a consensus validator?

Reply:
For example, we remove a chain maintainer when the proxy is deactivated, but not if the operator unbonds as a validator while never deactivating their proxy account.

Reply:
It is a separate concept.